src/include/misc.i

```diff
@@ -70,3 +70,19 @@ __wt_verbose(WT_SESSION_IMPL *session, int flag, const char *fmt, ...)
 	return (0);
 #endif
 }
+
+/*
+ * __wt_diagnostic_yield --
```

Comment:
I'd also like to have a diagnostic function to introduce aborts or errors in some code paths.  We could generalize this to support those uses too or add a couple more similar functions.